src/main/java/Duke.java

```diff
@@ -1,10 +1,139 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    // Constants that store index no. of "deadline" and "event"
+    public static final int DASH_INDX     = 4;
+    public static final int TODO_SIZE     = 5;
+    public static final int EVENT_SIZE    = 6;
+    public static final int DEADLINE_SIZE = 9;
+    public static final Task[] tasks      = new Task[100];
+    public static final String LINE       = "--------------------------------------------------------------------------------";
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+
+        // Prints logo and welcome message
+        printWelcome();
+
+        int itemIndex = 0;
+        boolean saidBye = false;
+
+        Scanner in = new Scanner(System.in);
+
+        // Cond. is true when input "bye" is given, terminates run
+        while (!saidBye) {
+
+            // Takes in user commands to interact with bot
+            String command = in.nextLine();
+
+            if (command.equals("bye")) {
+                saidBye = true;
+                printBye();
+            } else if (command.equals("list")) {
+                printList(tasks, itemIndex);
+            } else if (command.contains("done")) {
+                doneMethod(tasks, itemIndex, command);
+            } else if (command.equals("echo")) {
+                echoMode();
+            } else {
+                // Create tasks for class Todo
+                if (command.contains("todo")) {
+                    // Reads substring containing the task description only
+                    tasks[itemIndex] = new Todo(command.substring(TODO_SIZE));
+                }
+                // Create tasks for class Deadline
+                else {
+                    final String dateOrTime = command.substring(command.indexOf("/") + DASH_INDX);
```

Comment:
Consider removing the final, it is not required here. Constant should be specified at the beginning of the class.